### PR TITLE
Added Virtualmachine count and ID's to listSecurityGroups response

### PR DESCRIPTION
**Pull request by borisroman**
See issue CLOUDSTACK-8133 for more information.

### CODE REVIEW

File: server/src/com/cloud/api/query/dao/SecurityGroupJoinDaoImpl.java [hunk 1]
```diff
@@ -125,6 +130,14 @@ public SecurityGroupResponse newSecurityGroupResponse(SecurityGroupJoinVO vsg, A
             }
         }
 
+        List<SecurityGroupVMMapVO> securityGroupVmMap = _securityGroupVMMapDao.listBySecurityGroup(vsg.getId());
+        s_logger.debug("newSecurityGroupResponse() -> virtualmachine count: " + securityGroupVmMap.size());
+        sgResponse.setVirtualMachineCount(securityGroupVmMap.size());
+
+        for(SecurityGroupVMMapVO securityGroupVMMapVO : securityGroupVmMap) {
+            sgResponse.addVirtualMachineId(_userVmDao.findById(securityGroupVMMapVO.getInstanceId()).getUuid());
```
Comment (koushik-das) [inline server/src/com/cloud/api/query/dao/SecurityGroupJoinDaoImpl.java]: Add a check that findById is not returning null. There is a possibility that VM may get deleted after the call to listBySecurityGroup() but before that map entry is processed in the for-loop.

**Comment by koushik-das** [comment]
Please add tests.

**Comment by borisroman** [comment]
@wilderrodrigues @DaanHoogland @remibergsma Here are the interface changes related to https://github.com/apache/cloudstack/pull/593.

**Comment by borisroman** [comment]
@koushik-das Added tests. Please have a look!

File: server/src/com/cloud/api/query/dao/SecurityGroupJoinDaoImpl.java [hunk 2]
```diff
@@ -125,6 +130,15 @@ public SecurityGroupResponse newSecurityGroupResponse(SecurityGroupJoinVO vsg, A
             }
         }
 
+        List<SecurityGroupVMMapVO> securityGroupVmMap = _securityGroupVMMapDao.listBySecurityGroup(vsg.getId());
+        s_logger.debug("newSecurityGroupResponse() -> virtualmachine count: " + securityGroupVmMap.size());
+        sgResponse.setVirtualMachineCount(securityGroupVmMap.size());
+
+        for(SecurityGroupVMMapVO securityGroupVMMapVO : securityGroupVmMap) {
+            if (_userVmDao.findById(securityGroupVMMapVO.getInstanceId()) != null)
```
Comment (wido) [inline server/src/com/cloud/api/query/dao/SecurityGroupJoinDaoImpl.java]: Could you enclose this if by brackets? { }

**Comment by wido** [comment]
LGTM if you fix the brackets in the if-statement

**Comment by remibergsma** [comment]
@borisroman Could you please squash the commits into 1 (or maybe 2 if you want the tests separate).

File: server/src/com/cloud/api/query/dao/SecurityGroupJoinDaoImpl.java [hunk 3]
```diff
@@ -125,6 +130,16 @@ public SecurityGroupResponse newSecurityGroupResponse(SecurityGroupJoinVO vsg, A
             }
         }
 
+        List<SecurityGroupVMMapVO> securityGroupVmMap = _securityGroupVMMapDao.listBySecurityGroup(vsg.getId());
+        s_logger.debug("newSecurityGroupResponse() -> virtualmachine count: " + securityGroupVmMap.size());
+        sgResponse.setVirtualMachineCount(securityGroupVmMap.size());
+
+        for(SecurityGroupVMMapVO securityGroupVMMapVO : securityGroupVmMap) {
+            if (_userVmDao.findById(securityGroupVMMapVO.getInstanceId()) != null) {
```
Comment (koushik-das) [inline server/src/com/cloud/api/query/dao/SecurityGroupJoinDaoImpl.java]: @borisroman Please make findById() call only once. Something like

UserVMVO vm = _userVmDao.findById(securityGroupVMMapVO.getInstanceId());
if (vm != null) { sgResponse.addVirtualMachineId(vm.getUuid()); }

**Comment by koushik-das** [comment]
@borisroman I have provided a code comment. Once you fix that and squash the commits as suggested by @remibergsma then its a LGTM from me. Thanks for the unit tests.

**Comment by borisroman** [comment]
@wido @remibergsma @koushik-das Folowed up on lastest comments by Remi and Koushik.

**Pull request closed without merging**